### ISSUE TITLE
Automatic Detect Crashing

### ISSUE THREAD

**Issue by chappy1978** [vonnieda/AVRFuses#5]
It seems there is a bug in auto detect that causes AVRFuses to crash on a Sierra. This is a bit from the crash report. I tried building fuses in XCode with no luck. Xcode said these lines from AVRFusesController.m is where there is a problem. The error is EXC_BAD_ACCESS (code 1, address 0x08)

```
[[NSUserDefaults standardUserDefaults] setObject: selectedPart->name forKey: @"lastSelectedPart"];

[self deviceChanged: nil];
```
	
 If you need more information feel free to ask.

Process:               AVRFuses [9685]
Path:                  /Applications/Arduino/AVRFuses.app/Contents/MacOS/AVRFuses
Identifier:            org.vonnieda.AVRFuses
Version:               1.4.12
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           AVRFuses [9685]
User ID:               501

Date/Time:             2017-03-09 21:02:40.940 -0800
OS Version:            Mac OS X 10.12.4 (16E183b)
Report Version:        12
Anonymous UUID:        1137D4D2-A547-7D18-5C39-39D7919E06F1

Sleep/Wake UUID:       EAE1C564-519C-4507-9649-C1A5ECA22CB3

Time Awake Since Boot: 19000 seconds
Time Since Wake:       10000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000008
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]

**Comment by chappy1978** [comment]
A little more info from XCode
~/AVRFuses-master/AVRFusesController.m:426:10: 'beginSheetModalForWindow:modalDelegate:didEndSelector:contextInfo:' is deprecated: first deprecated in macOS 10.10 - Use -beginSheetModalForWindow:completionHandler: instead

**Comment by vonnieda** [comment]
Thanks for the report @chappy1978. @trol73, would you be able to look into this?

**Comment by trol73** [comment]
How can I reproduce this?
I'm trying to press Autodetect on Sierra but I don't get any crash

**Comment by chappy1978** [comment]
I looked it over last night. It seems a pointer is calling a deallocated location in memory after it calls avrdude for autodetect. I'm not a objective c guy so I been really brought to a stand still.

**Comment by chappy1978** [comment]
Do you have a programmer and a avr connected to the USB. It only happens when avrdude has successfully called for a signature.

**Comment by trol73** [comment]
Yes, I have usbasp with connected atmega128a / atmega328p. And avrdude has successfully detected for a signature..  On Sierra 10.12.3 and El Capitan 10.11.6 I don't see any crashes..

**Comment by vonnieda** [comment]
@trol73 Are you using 1.4.12? Or the latest code? Let's make sure everyone is using the same version.

**Comment by trol73** [comment]
I'm using 1.4.12 downloaded from your site.

**Comment by chappy1978** [comment]
What version of avrdude? I'm using 6.3

**Comment by trol73** [comment]
I'm using 6.3 too

**Comment by chappy1978** [comment]
AVRFuse 1.4.12
Avrdude 6.3
Sierra newest beta 
Usbasp newest
At tiny 85

**Comment by trol73** [comment]
Looks like that microcontroller name make sense here. ATtiny85 is not in the list Signatures.lst

**Comment by trol73** [comment]
I have reproduced that - when I removed device from Signatures.lst I get crash

**Comment by trol73** [comment]
I have fixed that. Sorry for the issue, I'm not a obj-c programmer and this is my only second micro project :)

**Comment by chappy1978** [comment]
Hey no problem. I'll check out the changes on Monday. Thanks
It's a handy little piece of code you made.

**Comment by chappy1978** [comment]
I added the new code and tested it out without the attiny signature and it work it returns
SUCCESS
(null) 
But not the @"Device not found"
Also I added the attiny85 to the signature list and that worked great. Is there a extended signature list?
I found many of the signatures could be extracted from the avrdude.config 
Thought?

**Comment by vonnieda** [comment]
Thanks for the fix @trol73. It has been released in 1.4.13.
